### PR TITLE
CRC big endian support

### PR DESCRIPTION
**Pull request by DmitriyMusatkin**
*Issue #, if available:*

*Description of changes:*
big endian support for crc algos + ci

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/aws/checksums/private/crc_util.h [hunk 1]
```diff
@@ -21,4 +22,32 @@
...
+    if (!aws_is_big_endian()) {
+        return x;
+    }
+
+    uint64_t m;
+    m = 0xff00ff00ff00ff;
+    x = ((x >> 8) & m) | (x & m) << 8;
+    m = 0xffff0000ffff;
+    x = ((x >> 16) & m) | (x & m) << 16;
+    return x >> 32 | x << 32;
```
Comment (graebm) [inline include/aws/checksums/private/crc_util.h:50]: if this algorithm came from somewhere, include a link

Comment (graebm) [reply include/aws/checksums/private/crc_util.h:50]: I worked it out on paper just now ... 😮‍💨 seems right

File: include/aws/checksums/private/crc_util.h [hunk 2]
```diff
@@ -21,4 +22,32 @@
         return val;                                                                                                    \
     }
 
+/* helper function to reverse byte order on big-endian platforms*/
+static inline uint32_t aws_swap_bytes_if_needed_32(uint32_t x) {
```
Comment (graebm) [inline include/aws/checksums/private/crc_util.h:26]: trivial: maybe I'm just not familiar with the literature, but to me "swap_bytes_if_needed" doesn't indicate which way they would swap, or what would be needed to make them swap
```suggestion
static inline uint32_t aws_swap_bytes_to_le_if_needed_32(uint32_t x) {
```

File: include/aws/checksums/private/crc_util.h (hunk 1 above)
Comment (graebm) [reply include/aws/checksums/private/crc_util.h:50]: if no link, add comments explaining like
// swap every other byte, now every 2-byte chunk is reversed
...
// swap every other 2-byte chunk, now every 4-byte chunk is reversed
...
// swap the two 4-byte chunks, now all 8 bytes are reversed

File: include/aws/checksums/private/crc_util.h [hunk 3]
```diff
@@ -21,4 +22,32 @@
...
+    if (!aws_is_big_endian()) {
+        return x;
+    }
+
+    uint8_t c1 = x & 0xFF;
+    uint8_t c2 = (x >> 8) & 0xFF;
+    uint8_t c3 = (x >> 16) & 0xFF;
+    uint8_t c4 = (x >> 24) & 0xFF;
+
+    return ((uint32_t)c1 << 24) + ((uint32_t)c2 << 16) + ((uint32_t)c3 << 8) + c4;
```
Comment (graebm) [inline include/aws/checksums/private/crc_util.h:36]: debatable:
this is so much easier to follow than the 64-bit algorithm
should this one be trickier?
or should the other one be simpler?

File: tests/crc_test.c [hunk 4]
```diff
@@ -73,8 +73,9 @@ static int s_test_known_crc_32(
     uint32_t result = func(input, (int)length, 0);
     ASSERT_HEX_EQUALS(expected_crc, result, "%s(%s)", func_name, data_name);
 
+    uint32_t result_swapped = aws_swap_bytes_if_needed_32(result);
```
Comment (graebm) [inline tests/crc_test.c:76]: trivial: it's not super obvious which way "swapped" is
```suggestion
    uint32_t result_little_endian = aws_swap_bytes_if_needed_32(result);
```
or
```suggestion
    uint32_t result_le = aws_swap_bytes_if_needed_32(result);
```

File: source/crc64_sw.c [hunk 5]
```diff
@@ -538,18 +540,580 @@ static uint64_t crc64nvme_table[8][256] = {
...
+        crc = (crc << 8) ^ crc64nvme_table_be[0][((crc >> 56) ^ *input++)];
+        length--;
+    }
+
+    int remaining = length;
+    // Once we are aligned, read 8 bytes at a time
+    const uint64_t *current = (const uint64_t *)(const void *)input;
+    while (remaining >= 8) {
+        uint64_t c1 = *current++ ^ crc;
+        crc = crc64nvme_table_be[0][c1 & 0xff] ^ crc64nvme_table_be[1][(c1 >> 8) & 0xff] ^
```
Comment (graebm) [inline source/crc64_sw.c:1088]: I'm not smart enough to check this in under a day ... but it looks like the other one, and the tests pass so probably good 👍

**Review by graebm — APPROVED** [review]
lots of trivial stuff, but looks good

File: include/aws/checksums/private/crc_util.h (hunk 1 above)
Comment (DmitriyMusatkin) [reply include/aws/checksums/private/crc_util.h:50]: This bit twiddling trick is pretty popular in fast crc implementations in the wild. Don't know where it originated, but most libs i looked at included a variant of this. 

Poking around more modern hash implementations, like xxHash, that trick seems to have fallen out of favor. Replaced it with compiler built ins and a typical posix byte swap implementation as fallback.

File: include/aws/checksums/private/crc_util.h (hunk 2 above)
Comment (DmitriyMusatkin) [reply include/aws/checksums/private/crc_util.h:26]: renamed to aws_bswap64/32_if_be. 
swap bytes or byte swap (bswap is common intrinsic name) is a fairly typical name for reversing order of all bytes.

File: include/aws/checksums/private/crc_util.h (hunk 3 above)
Comment (DmitriyMusatkin) [reply include/aws/checksums/private/crc_util.h:36]: i used version from original pr. intrinsics should be a bit faster here and this is in somewhat of a hot path.

**Pull request merged**